File: matrix/number_of_laser_beam_in_bank.py
from typing import List

class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        ROWS, COLS = len(bank), len(bank[0])
        laser_cell = {}
        
        for r in range(ROWS):
            laser_cell[r] = 0
            for c in range(COLS):
                if bank[r][c] == '1':
                    laser_cell[r] += 1
        
        total_beams = 0

        for r in range(ROWS - 1):
            for nextRow in range(r + 1, ROWS):
                if laser_cell[r] > 0 and laser_cell[nextRow] > 0:
                    total_beams += laser_cell[r] * laser_cell[nextRow]
                    break

        return total_beams
            

File: matrix/number_of_laser_beam_in_bank.java
package matrix;

import java.util.HashMap;
import java.util.Map;

public class number_of_laser_beam_in_bank {
    
}

class Solution {
    public int numberOfBeams(String[] bank) {
        int ROWS = bank.length, COLS = bank[0].length();
        Map<Integer, Integer> laserBeams = new HashMap<>();

        for(int r = 0; r < ROWS; r++){
            laserBeams.put(r, 0);
            for(int c = 0; c < COLS; c++){
                if(bank[r].charAt(c) == '1'){
                    int val = laserBeams.get(r) + 1;
                    laserBeams.put(r, val);
                }
            }
        }

        int totalBeams = 0;
        for(int row = 0; row < ROWS; row++){
            for(int nextRow = row + 1; nextRow < ROWS; nextRow++){
                if(laserBeams.get(row) > 0 && laserBeams.get(nextRow) > 0){
                    totalBeams += laserBeams.get(row) * laserBeams.get(nextRow);
                    break;
                }
            }
        }

        return totalBeams;

    }
}